src/declarations/stencil-public-runtime.ts

```diff
@@ -1257,6 +1257,8 @@ export namespace JSXBase {
   }
 
   export interface TextareaHTMLAttributes<T> extends HTMLAttributes<T> {
+    autoComplete?: 'on' | 'off';
+    autocomplete?: 'on' | 'off';
```

Comment:
I'm a little unsure of exactly what's the standard behavior here. The [MDN docs for <textarea>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attributes) imply that `'on'` and `'off'` are the only supported values for `autocomplete` on a `<textarea>`, but the [MDN page for `autocomplete`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) shows a much wider range of possible values like `"given-name"`, `"email"`, etc.

I went at looked at [the whatwg standard document](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofilling-form-controls:-the-autocomplete-attribute) and I haven't found something explicitly saying 'these values should be allowed on a `<textarea>`' _but_ some of the examples show other values being passed to autocomplete, like in this example code:

```html
<fieldset>
 <legend>Ship the blue gift to...</legend>
 <p> <label> Address:     <textarea name=ba autocomplete="section-blue shipping street-address"></textarea> </label>
 <p> <label> City:        <input name=bc autocomplete="section-blue shipping address-level2"> </label>
 <p> <label> Postal Code: <input name=bp autocomplete="section-blue shipping postal-code"> </label>
</fieldset>
<fieldset>
 <legend>Ship the red gift to...</legend>
 <p> <label> Address:     <textarea name=ra autocomplete="section-red shipping street-address"></textarea> </label>
 <p> <label> City:        <input name=rc autocomplete="section-red shipping address-level2"> </label>
 <p> <label> Postal Code: <input name=rp autocomplete="section-red shipping postal-code"> </label>
</fieldset>
```

So anyway this is a long way to say we might have to worry about a wider type here - perhaps we should just do `string` and let the browser worry about doing the right thing? Alternatively we could maintain a string union type that captures all these possible values, like

```ts
type AutocompleteOption =
  | "on"
  | "off"
  | "email"
  | "cc-number"
  | ... ; // etc
```

Reply:
I would recommend a `string` type to align with the implementation for `HTMLInputElement`, etc.: https://github.com/TypeStrong/tscs/blob/master/lib/node_modules/TypeScript/bin/lib.dom.d.ts#L6814-L6817

Reply:
Agreed. Should keep it in line with the `HTMLInputElement` spec. Better to make it wider rather than receive another bug report about the type being too narrow.